### PR TITLE
Update Dart Analysis Server sources.

### PR DESCRIPTION
**Pull request by pq**
* picks up added `defaultArgumentListString` property in the new `CompletionSuggestion` API

@alexander-doroshko

### CODE REVIEW

File: Dart/thirdPartySrc/analysisServer/org/dartlang/analysis/server/protocol/AnalysisError.java [hunk 1]
```diff
@@ -122,7 +122,7 @@ public static AnalysisError fromJson(JsonObject jsonObject) {
     Location location = Location.fromJson(jsonObject.get("location").getAsJsonObject());
     String message = jsonObject.get("message").getAsString();
     String correction = jsonObject.get("correction") == null ? null : jsonObject.get("correction").getAsString();
-    String code = jsonObject.get("code") == null ? null : jsonObject.get("code").getAsString();
+    String code = jsonObject.get("code").getAsString();
```
Comment (alexander-doroshko) [inline Dart/thirdPartySrc/analysisServer/org/dartlang/analysis/server/protocol/AnalysisError.java]: This will throw NPE with older SDKs

Comment (pq) [inline Dart/thirdPartySrc/analysisServer/org/dartlang/analysis/server/protocol/AnalysisError.java]: OK, will fix.

**Comment by pq** [comment]
It looks like I missed a few API changes...  Stand-by for update.

**Comment by alexander-doroshko** [comment]
The rest looks safe.
I guess these sources are generated. Is it possible to fix anything somewhere to make it generate backward-compatible code?

**Comment by pq** [comment]
> It looks like I missed a few API changes... Stand-by for update.

Done.

> Is it possible to fix anything somewhere to make it generate backward-compatible code?

Great question.

cc @scheglov @bwilkerson @jwren for thoughts.

**Comment by bwilkerson** [comment]
> Is it possible to fix anything somewhere to make it generate backward-compatible code?

Maybe. We only have one version of the spec to work from when generating, so that will limit us to some extent, but depending on what you need to do it might be enough.

There is also an opportunity to have it generate more Java code so that we need to do less when the spec is updated. Jaime and I discussed this when we were first writing the generator, but the work was not finished.

The code to generate the Java support code is in these two files:
https://github.com/dart-lang/sdk/blob/master/pkg/analysis_server/tool/spec/codegen_java.dart
https://github.com/dart-lang/sdk/blob/master/pkg/analysis_server/tool/spec/codegen_java_types.dart

**Comment by alexander-doroshko** [comment]
> Is it possible to fix anything somewhere to make it generate backward-compatible code?

For the reference: I was asking about [this code](https://github.com/dart-lang/sdk/blob/66b5248bf937ef1202f04b106c63730e57a008b8/pkg/analysis_server/tool/spec/generated/java/types/RequestError.java#L83) that throws NPE with older SDKs
```
  public static RequestError fromJson(JsonObject jsonObject) {
    String code = jsonObject.get("code").getAsString();
    ...
  }
```

**Comment by bwilkerson** [comment]
Then, yes, it should be possible to fix the way it generates field accesses.

**Comment by devoncarew** [comment]
Perhaps just marking `code` as optional?

**Comment by bwilkerson** [comment]
The field 'code' is not optional and it never has been. Marking it as optional would be the wrong solution (and a breaking change to the API).

@alexander-doroshko Have we run into a situation where server sent back a RequestError that did not have a code associated with it? If so, we need to fix the server. Either way, should we just always be defensive (always check for null, even when the field is required)?

**Comment by alexander-doroshko** [comment]
@bwilkerson Sorry, I gave a wrong link. I meant [AnalysisError.code](https://github.com/dart-lang/sdk/blob/66b5248bf937ef1202f04b106c63730e57a008b8/pkg/analysis_server/tool/spec/generated/java/types/AnalysisError.java#L125) introduced in [this commit](https://github.com/dart-lang/sdk/commit/e7e205f00b24faf9ee5cdf0baf21b8a0a04f8c3a). Older SDKs do not have analysis error code.

**Comment by bwilkerson** [comment]
Ok. Then for backward compatibility we should always check for null, even for required fields, because they might not have been required in previous versions. That seems reasonable to me.

**Comment by alexander-doroshko** [comment]
We've already had [this discussion](https://github.com/JetBrains/intellij-plugins/pull/399/files/efe77c62f84dd2fcfe71426d4d5f40cba8a2ba55#r75862948) half a year ago and at that point decided just to stay with a couple of manually modified lines :).

**Comment by bwilkerson** [comment]
I had forgotten that discussion. However, it might be worth reconsidering. Seems like that decision is leading to errors and that it might be safer to modify the generator. But I'm fine with either decision.

**Comment by alexander-doroshko** [comment]
Merged

**Comment by pq** [comment]
Thanks!

**Pull request closed without merging**